internal/blockchain/ethereum/ethereum.go

```diff
@@ -106,13 +106,19 @@ func (cb *callbacks) BlockchainNetworkAction(ctx context.Context, namespace, act
 	return nil
 }
 
-func (cb *callbacks) BlockchainEvent(event *blockchain.EventWithSubscription) error {
-	for _, cb := range cb.handlers {
-		// Send the event to all handlers and let them match it to a contract listener
-		// TODO: can we push more listener/namespace knowledge down to this layer?
-		if err := cb.BlockchainEvent(event); err != nil {
-			return err
+func (cb *callbacks) BlockchainEvent(ctx context.Context, event *blockchain.EventWithSubscription, namespace string) error {
```

Comment:
nit: can we order `namespace` right after `ctx` like on the other methods above?